File: _app/src/main/java/demo/com/demolibraries/main/MenuAdapter.java
package demo.com.demolibraries.main;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import demo.com.demolibraries.R;

public class MenuAdapter extends RecyclerView.Adapter<MainViewHolder> {
    private List<MenuEntity> data;
    private OnItemClickListener onItemClickListener;

    public MenuAdapter(@Nullable List<MenuEntity> menuEntities, OnItemClickListener onItemClickListener) {
        if (menuEntities == null) {
            data = new ArrayList<>();
        } else data = menuEntities;

        this.onItemClickListener = onItemClickListener;
    }


    @NonNull
    @Override
    public MainViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_menu, parent, false);
        return new MainViewHolder(v, onItemClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull MainViewHolder holder, int position) {
        holder.bindView(data.get(position));
        holder.showDivder(position < getItemCount() - 1);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }
}

File: _app/src/main/java/demo/com/demolibraries/main/MenuAdapter.kt
package demo.com.demolibraries.main

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import java.util.ArrayList

import demo.com.demolibraries.R

class MenuAdapter(menuItems: List<MenuItem>?, private val onItemClickListener: OnItemClickListener) : RecyclerView.Adapter<MainViewHolder>() {
    private var data: List<MenuItem>? = null

    init {
        data = menuItems ?: ArrayList()
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_menu, parent, false)
        return MainViewHolder(v, onItemClickListener)
    }

    override fun onBindViewHolder(holder: MainViewHolder, position: Int) {
        holder.bindView(data!![position])
        holder.showDivider(position < itemCount - 1)
    }

    override fun getItemCount(): Int {
        return data!!.size
    }
}
